lib/utils/src/graph/algorithms.cc

```diff
@@ -498,4 +592,45 @@ MultiDiGraphView as_multidigraph(OpenMultiDiGraphView const &g) {
   return MultiDiGraphView::create<ViewOpenMultiDiGraphAsMultiDiGraph>(g);
 }
 
+std::vector<std::unordered_set<Node>>
+    get_weakly_connected_components(DiGraphView const & g) {
+    std::unordered_set<Node> start_pointes = get_sources(g);
+    std::vector<Node>  dfs_order = get_dfs_ordering(g, start_pointes);
+
+    std::vector<std::unordered_set<Node>> components;
+    std::unordered_set<Node> visited;
+
+    for (const auto& node : dfs_order) {
```

Comment:
This should just be implemented in terms of an operation on `get_connected_components(UndirectedGraphView const &)`

Reply:
Yes (though I think the first may already be implemented--you should check)

Reply:
I found there is no method for `get_weakly_connected_components` that is implemented